provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright (c) 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trusted_ip_addresses
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+)
+
+type TrustedIpsDataSource struct {
+	collection *cmv1.TrustedIpsClient
+}
+
+var _ datasource.DataSource = &TrustedIpsDataSource{}
+var _ datasource.DataSourceWithConfigure = &TrustedIpsDataSource{}
+
+func New() datasource.DataSource {
+	return &TrustedIpsDataSource{}
+}
+
+func (s *TrustedIpsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_trusted_ip_addresses"
+}
+
+func (s *TrustedIpsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "List of trusted ip addresses",
+		Attributes: map[string]schema.Attribute{
+			"page": schema.Int64Attribute{
+				Description: "Page number of the result set.",
+				Computed:    true,
+			},
+			"size": schema.Int64Attribute{
+				Description: "Number of items per page.",
+				Computed:    true,
+			},
```

Comment:
Its not clear to me why the pagination info should be exposed to the user.
This is internal implementation info and the user shouldn't have this in the output.
What do you think?

Reply:
I'll update to just include the total and Items